### ISSUE TITLE
java.util.ConcurrentModificationException in EditProfileActivity

### ISSUE THREAD

**Issue by karanatwal** [chat-sdk/chat-sdk-android#194]
While updating status i am getting crash due to ConcurrentModificationException at `i.hasNext()`

```
 Iterator<String> i = currentUser.metaMap().keySet().iterator();
        while (i.hasNext()) {
            String key = i.next();
            if(key.equals(Keys.AvatarURL)) {
                imageChanged = valueChanged(currentUser.metaMap(), userMeta, key);
                currentUser.setAvatarHash(null);
            }
            if(key.equals(Keys.Availability) || key.equals(Keys.Status)) {
                presenceChanged = presenceChanged || valueChanged(currentUser.metaMap(), userMeta, key);
            }
        }
```

**Comment by bensmiley** [comment]
Could you post the stack trace?

**Comment by bensmiley** [comment]
I managed to replicate this issue and it's fixed in version [4.0.6](https://github.com/chat-sdk/chat-sdk-android/releases/tag/4.0.6).
